src/main/java/com/olgaboiar/mint/handlers/OptionsHandler.java

```diff
@@ -1,25 +1,16 @@
 package com.olgaboiar.mint.handlers;
 
-import com.olgaboiar.mint.Request;
-import com.olgaboiar.mint.Response;
-import com.olgaboiar.mint.ResponseGenerator;
+import com.olgaboiar.mint.*;
 
 import java.io.IOException;
-
-import static com.olgaboiar.mint.Constants.STATUS_CODE_200;
+import java.util.ArrayList;
 
 public class OptionsHandler implements IHandler {
-    String[] allowedMethods;
-    Response response;
-
-    public OptionsHandler(String[] allowedMethods) {
-        this.allowedMethods = allowedMethods;
-    }
 
     @Override
-    public Response handleRequest(Request request) throws IOException {
-        String body = "";
-        response =  new ResponseGenerator().generateResponse(STATUS_CODE_200, body);
+    public Response handleRequest(Request request, IRouteMap routes) throws IOException {
+        Response response =  new ResponseGenerator().generateResponse(Constants.Status.STATUS_CODE_200.toString(), "");
+        ArrayList<String> allowedMethods = routes.getAllowedMethods(request.getUri());
```

Comment:
Rather than calling #toString on the status code each time you call #generateResponse, perhaps you can pass in the status code enum to the response generator and the header generator, and call #toString right before you need it.

Reply:
refactored in https://github.com/olgaboiar/MintHTTPServer/pull/10/commits/6f26af88a680fb4f98f68d972c984b0b38fcbd55